### ISSUE TITLE
Error "Cannot parse request" when creating new VM with new Vnet and Ip address

### ISSUE THREAD

**Issue by travismc1** [Azure/azure-sdk-for-net#3385]
I'm unable to create a new VM using the Fluent SDK following the example here:
https://github.com/Azure-Samples/compute-dotnet-manage-virtual-machine-using-vm-extensions/blob/master/Program.cs

Whenever I run this portion of the code
```
IVirtualMachine linuxVM = azure.VirtualMachines.Define("travis616test")
      .WithRegion(Region.EuropeNorth)
      .WithExistingResourceGroup("travis616rg2")
      .WithNewPrimaryNetwork("10.0.0.0/28")
      .WithPrimaryPrivateIPAddressDynamic()
      .WithNewPrimaryPublicIPAddress(pipDnsLabelLinuxVM)
      .WithPopularLinuxImage(KnownLinuxVirtualMachineImage.UbuntuServer16_04_Lts)
      .WithRootUsername(FirstLinuxUserName)
      .WithRootPassword(FirstLinuxUserPassword)
      .WithSize(VirtualMachineSizeTypes.StandardDS1V2)
      .Create();
```

it fails with "Cannot parse request"
`Microsoft.Rest.Azure.CloudException: Cannot parse the request.
   at Microsoft.Azure.Management.Network.Fluent.NetworkInterfacesOperations.<BeginCreateOrUpdateWithHttpMessagesAsync>d__16.MoveNext()

This is using the latest Fluent nuget packages: 1.1.1

**Comment by martinsawicki** [comment]
I tried the code above and I did not run into any problems.
Can you please check the string parameters passed into this code are valid/not empty/not null, etc?
The above code uses a number of variables, but the values are not provided here. 
So I used literal strings to try to reproduce and had no issues. 

```csharp
                azure.ResourceGroups.Define("testgroup")
                    .WithRegion(Region.EuropeNorth)
                    .Create();

                IVirtualMachine testVM = azure.VirtualMachines.Define("travis616test")
                    .WithRegion(Region.EuropeNorth)
                    .WithExistingResourceGroup("testgroup")
                    .WithNewPrimaryNetwork("10.0.0.0/28")
                    .WithPrimaryPrivateIPAddressDynamic()
                    .WithNewPrimaryPublicIPAddress("test616test")
                    .WithPopularLinuxImage(KnownLinuxVirtualMachineImage.UbuntuServer16_04_Lts)
                    .WithRootUsername("tester")
                    .WithRootPassword("Abcdef.123456!")
                    .WithSize(VirtualMachineSizeTypes.StandardDS1V2)
                    .Create();
```

**Comment by travismc1** [comment]
I still receive the error, but maybe this will help.

I provided the values inline to assist.

            azure.ResourceGroups.Define("travis619rg")
                                .WithRegion(Region.EuropeNorth)
                               .Create();

            IVirtualMachine testVM = azure.VirtualMachines.Define("travis619test")
                .WithRegion(Region.EuropeNorth)
                .WithExistingResourceGroup("travis619rg")
                .WithNewPrimaryNetwork("10.0.0.0/28")
                .WithPrimaryPrivateIPAddressDynamic()
                .WithNewPrimaryPublicIPAddress("travis619ip")
                .WithPopularLinuxImage(KnownLinuxVirtualMachineImage.UbuntuServer16_04_Lts)
                .WithRootUsername("travismc")
                .WithRootPassword("@LinuxIsv123!")
                .WithSize(VirtualMachineSizeTypes.StandardDS1V2)
                .Create();

When it encountered the error, the exception object had this as the request payload:

   {
  "properties": {
    "ipConfigurations": [
      {
        "properties": {
          "privateIPAllocationMethod": "Dynamic",
          "subnet": {
            "id": "/subscriptions/b835f458-560c-49c2-8781-4b13d892ddd9/resourceGroups/travis619rg/providers/Microsoft.Network/virtualNetworks/vnet8fc50141dbb4/subnets/subnet1"
          },
          "publicIPAddress": {
            "properties": {
              "publicIPAllocationMethod": "Dynamic",
              "publicIPAddressVersion": "IPv4",
              "dnsSettings": {
                "domainNameLabel": "travis619ip",
                "fqdn": "travis619ip.northeurope.cloudapp.azure.com"
              },
              "idleTimeoutInMinutes": 4,
              "resourceGuid": "d96e6747-844c-4a81-92e1-414376a84366",
              "provisioningState": "Succeeded"
            },
            "etag": "W/\"8051bc44-7148-40e4-b01f-a6d1260f7faf\"",
            "location": "northeurope",
            "tags": {}
          }
        },
        "name": "primary"
      }
    ],
    "dnsSettings": {}
  },
  "location": "northeurope",
  "tags": {}
   }

I made an ARM template with those properties and received the following response.
`{
  "error": {
    "code": "InvalidRequestFormat",
    "message": "Cannot parse the request.",
    "details": [
      {
        "code": "MissingJsonReferenceId",
        "message": "Value for reference id is missing. Path properties.ipConfigurations[0].properties.publicIPAddress."
      }
    ]
  }
}`

**Comment by hovsepm** [comment]
@travismc1 This should be fixed in v1.1.2. Please update your reference and post here results.

**Comment by hovsepm** [comment]
this should have been fixed in v 1.1.2. Please re-open as a new bug if it still repros
